src/Microsoft.ML.FastTree/TreeEnsembleFeaturizer.cs

```diff
@@ -715,16 +717,16 @@ public static IDataTransform CreateForEntryPoint(IHostEnvironment env, Arguments
                 // be non-null.
                 var vm = predictor as IValueMapper;
                 ch.CheckUserArg(vm != null, nameof(args.PredictorModel), "Predictor does not have compatible type");
-                if (data != null && vm?.InputType.VectorSize != data.Schema.Feature.Type.VectorSize)
+                if (data != null && vm.InputType.VectorSize != data.Schema.Feature.Type.VectorSize)
```

Comment:
>if (data != null && vm.InputType.VectorSize != data.Schema.Feature.Type.VectorSize) [](start = 16, length = 83)

If someone were to fit in a predictor that did implement `IValueMapper` (which are most), but that is nevertheless not a tree, what do we expect would happen? From my reading of the code it would just apply a generic scorer. Should we at least verify, in some fashion, that it *is* the type of predictor we expect beyond being merely a `IValueMapper`?

Reply:
Hi Tom, thanks for reviewing this change. If the predictor is an IValueMapper but is not a tree, then the constructor of TreeEnsembleFeaturizerBindableMapper below will fail.

---
In reply to: [187707032](https://github.com/dotnet/machinelearning/pull/131#discussion_r187707032) [](ancestors = 187707032)